### PR TITLE
Module 2

### PR DESCRIPTION
**Pull request by DmitriyBabii**

### CODE REVIEW

**Comment by DmitriyBabii** [comment]
Not done yet

**Comment by DmitriyBabii** [comment]
Done

File: src/main/java/com/alevel/module2/models/Invoice.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    public int countTelevision() {
+        int count = 0;
+        for (Technics technics : technic) {
+            if (technics.getType().equals("Television")) {
+                count++;
+            }
+        }
+        return count;
+    }
```
Comment (Fabolo12) [inline src/main/java/com/alevel/module2/models/Invoice.java:42]:
```suggestion
 public long countDeviceByType(String type) {
        return Arrays.stream(technic)
                .filter(technics -> technics.getType().equals(type))
                .count();
    }
```

But the Invoice should not do it

File: src/main/java/com/alevel/module2/models/Invoice.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    public int getCountTechnic() {
+        return technic.length;
+    }
+
+    public int getPrice() {
+        if (sumPrice() > limit) {
+            type = TypeSale.WHOLESALE;
+        } else {
+            type = TypeSale.RETAIL;
+        }
```
Comment (Fabolo12) [inline src/main/java/com/alevel/module2/models/Invoice.java:53]: Why does the invoice type change in the "getPrice" method?

File: src/main/java/com/alevel/module2/models/Invoice.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        return type;
+    }
+
+    private int sumPrice() {
+        int sum = 0;
+        for (Technics technics : technic) {
+            sum += technics.getPrice();
+        }
+        return sum;
+    }
```
Comment (Fabolo12) [inline src/main/java/com/alevel/module2/models/Invoice.java]:
```suggestion
    private int sumPrice() {
        return Arrays.stream(technic)
                .mapToInt(Technics::getPrice)
                .sum();
    }
```

File: src/main/java/com/alevel/module2/models/Invoice.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+    private String outputTechnic() {
+        StringBuilder str = new StringBuilder();
+        for (Technics technics : technic) {
+            str.append(technics);
+        }
+        return str.toString();
+    }
```
Comment (Fabolo12) [inline src/main/java/com/alevel/module2/models/Invoice.java]:
```suggestion
    private String outputTechnic() {
        return Arrays.stream(technic)
                .map(Technics::toString)
                .collect(Collectors.joining(" "));
    }
```

File: src/main/java/com/alevel/module2/models/Invoice.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+            result.append(date[0])
+                    .append(":")
+                    .append(date[1]);
+        } catch (NullPointerException e) {
+            return this.date.toString();
+        }
+
+        return result.toString();
+    }
```
Comment (Fabolo12) [inline src/main/java/com/alevel/module2/models/Invoice.java:105]: Just read http://www.seostella.com/ru/article/2012/02/05/formatirovanie-daty-v-java.html

File: src/main/java/com/alevel/module2/models/Month.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package com.alevel.module2.models;
+
+public enum Month {
+    Jan(1), Feb(2), Mar(3), Apr(4),
+    May(5), June(6), July(7), Aug(8),
+    Sep(9), Oct(10), Nov(11), Dec(12);
+
+    public final int number;
```
Comment (Fabolo12) [inline src/main/java/com/alevel/module2/models/Month.java:8]: private

File: src/main/java/com/alevel/module2/models/Technics.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.Getter;
+
+@Getter
+abstract public class Technics {
+    protected String series;
+    protected Screen screenType;
+    protected int price;
+
+    public Technics() {
+    }
```
Comment (Fabolo12) [inline src/main/java/com/alevel/module2/models/Technics.java:12]: What for?

File: src/main/java/com/alevel/module2/models/Telephone.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    public Telephone() {
+    }
+
+    public Telephone(String series, String model, Screen screenType, int price) {
+        super(series, screenType, price);
+        this.model = model;
+        count++;
+    }
+
+    public static int getCount() {
```
Comment (Fabolo12) [inline src/main/java/com/alevel/module2/models/Telephone.java:19]: Method 'getCount()' is never used

File: src/main/java/com/alevel/module2/models/Television.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    public Television(String series, int diagonal, Screen screenType, String country, int price) {
+        super(series, screenType, price);
+        this.diagonal = diagonal;
+        this.country = country;
+        count++;
+    }
+
+    public static int getCount() {
```
Comment (Fabolo12) [inline src/main/java/com/alevel/module2/models/Television.java:19]: Method 'getCount()' is never used

**Review by Fabolo12 — APPROVED** [review]
Overall, the code is well written. Separately, a good point is the use of Enum, not strings.

But the logic of method "readFile" is very implicit, from the name it appears that the method only reads the file, but in fact, it also creates the Technique and writes the log.

10

**Pull request merged**